WooCommerce/Classes/ViewRelated/Products/Variations/Add Attributes/AddAttributeViewController.swift

```diff
@@ -0,0 +1,298 @@
+//
+private extension AddAttributeViewController {
+
+    /// Renders ghost placeholder product attributes.
+    ///
+    func displayGhostTableView() {
+        let placeholderProductAttributesPerSection = [3]
+        let options = GhostOptions(displaysSectionHeader: false,
+                                   reuseIdentifier: BasicTableViewCell.reuseIdentifier,
+                                   rowsPerSection: placeholderProductAttributesPerSection)
+        ghostTableView.displayGhostContent(options: options,
+                                           style: .wooDefaultGhostStyle)
+        ghostTableView.isHidden = false
+    }
+
+    /// Removes ghost  placeholder product attributes.
+    ///
+    func removeGhostTableView() {
+        tableView.reloadData()
+        ghostTableView.removeGhostContent()
+        ghostTableView.isHidden = true
+    }
+
+    /// Displays the Sync Error Notice.
+    ///
+    func displaySyncingErrorNotice() {
+        let notice = Notice(title: Localization.syncErrorMessage, feedbackType: .error, actionTitle: Localization.retryAction) { [weak self] in
+            self?.viewModel.performFetch()
+        }
+
+        ServiceLocator.noticePresenter.enqueue(notice: notice)
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension AddAttributeViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = viewModel.sections[indexPath.section].rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension AddAttributeViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return 44
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let leftText = viewModel.sections[section].header else {
+            return nil
+        }
+
+        let headerID = TwoColumnSectionHeaderView.reuseIdentifier
+        guard let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: headerID) as? TwoColumnSectionHeaderView else {
+            fatalError()
```

Comment:
If I understand correctly, this would stop execution also in production. Would it make sense to throw an assertion, if the reason it's here is to warn developers that a view is not properly registered?